clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -51,12 +52,22 @@ class AzureNetworkClient extends AzureBaseClient {
     def result = new ArrayList<AzureLoadBalancerDescription>()
 
     try {
-      this.getNetworkResourceProviderClient(creds).getLoadBalancersOperations().listAll().getLoadBalancers().each {item ->
+      def loadBalancers = this.getNetworkResourceProviderClient(creds).getLoadBalancersOperations().listAll().getLoadBalancers()
```

Comment:
sorry, did not notice the actual lines before my previous answer... In the case above we return the full list of the resources as part of one single call and therefore all should have the same "lastReadTime" value associated with. We later use that value to determine whether the resource is current or not when compared with a different read for the same resource. We need to get the current time after the API call succeeded rather then before.